src/librustdoc/clean/inline.rs

```diff
@@ -306,16 +306,15 @@ fn merge_attrs(
     attrs: Attrs<'_>,
     other_attrs: Option<Attrs<'_>>,
 ) -> clean::Attributes {
-    let cap = attrs.len() + other_attrs.map_or(0, |a| a.len());
-    let mut merged_attrs: Vec<ast::Attribute> = Vec::with_capacity(cap);
-    // If we have additional attributes (from a re-export),
+    let mut merged_attrs: Vec<ast::Attribute> = Vec::new();
+    // NOTE: If we have additional attributes (from a re-export),
     // always insert them first. This ensure that re-export
     // doc comments show up before the original doc comments
     // when we render them.
-    if let Some(a) = other_attrs {
-        merged_attrs.extend(a.iter().cloned());
+    if let Some(inner) = other_attrs {
+        merged_attrs = inner.to_vec();
     }
-    merged_attrs.extend(attrs.to_vec());
+    merged_attrs.extend_from_slice(attrs);
     merged_attrs.clean(cx)
```

Comment:
Ok, what about this idea, which preserve the order and never performs an unnecessary copy:

```rust
    let merged_attrs = if let Some(inner) = other_attrs {
        let mut both_attrs = inner.to_vec();
        both_attrs.extend_from_slice(attrs);
    } else {
        attrs.to_vec()
    };
```

Reply:
The reason I want to push for this is because it's very rare to have additional documentation on a re-export, but very common to have a _lot_ of documentation on the original item.

Reply:
I don't think it has much different since `Vec::new` has cap=0.
Also, your and my code reallocates when `other_attrs` is_some, IIUC.

Reply:
One reallocation is really cheap though, it's only reallocation in a loop that's slow (which this doesn't do).